### ISSUE TITLE
Could not open input file: <project root>/crunz

### ISSUE THREAD

**Issue by pridit** [lavary/crunz#176]
**Crunz version**:
v1.11.0-beta.1

**PHP version**:
7.2.5

**Operating system type and version**:
Ubuntu 18.04.1

**Description**  
Since upgrading to this beta release, calling `crunz schedule:run` results in `Could not open input file: <project root>/crunz`

Intermittently, it may also throw `No task found! Please check your source path.` as well without any configuration change, but I have not been able to reliably reproduce this one.

Both of these are not an issue for releases prior to this version.

**How to reproduce**  
Upgrade to v1.11.0-beta.1 or later and attempt to call `schedule:run`.

**Comment by PabloKowalczyk** [comment]
@pridit i need some details to reproduce, please provide paths relative to `<project root>`:
1. Location of config (`crunz.yml`) file
2. Value of `source` parameter in config file
3. Location of tasks directory
4. Location of `vendors` directory
5. Location of `crunz`

@kinoute Could you try to investigate this bug locally?

I think that this bug is caused by changes in https://github.com/lavary/crunz/pull/146, all paths are relative to `cwd`, not `crunz` bin file.

**Comment by kinoute** [comment]
The thing is, I can't reproduce it locally on My Mac. With an empty project, respecting the same project's tree and even importing my crunz.yml file, it works (with the beta or the rc1 version). 

IIRC, I even tried on my server with an empty project and it was working as well, that's why I didn't report it, thinking my project was the problem.

**Comment by kinoute** [comment]
Since @pridit said he wasn't able to reproduce the bug everytime, I made a loop to see if I could catch it:

![capture d ecran 2018-12-25 a 16 52 31](https://user-images.githubusercontent.com/623763/50424519-86ee6400-0865-11e9-9347-f2ca0fac6cda.png)

We can see that with the loop, I get the "No task error" every time. But running directly the script, It works. Could it be a problem about how schedule:run is called, by who and where?

When I use part of the cronjob entry to simulate the cronjob that should run every minute, I get the same error as well:

![capture d ecran 2018-12-25 a 16 57 24](https://user-images.githubusercontent.com/623763/50424545-30355a00-0866-11e9-9964-5d2e25bbcdc3.png)

**Comment by PabloKowalczyk** [comment]
> Could it be a problem about how schedule:run is called, by who and where?

I think the key here is **where**, after #146 all paths resolution is relative to `cwd` (current working directory). Could you paste what is the `cwd` in `for` loop?

About the second screenshot, try to do `* * * * * cd /Volumes/Storage/kinoute/Sites/crunz-error && vendor/bin/crunz schedulre:run`, if this works it means this bug is definitely related to paths resolution and PR #146.

**Comment by kinoute** [comment]
`* * * * * cd /Volumes/Storage/kinoute/Sites/crunz-error && vendor/bin/crunz schedule:run` works.

Also: 
![capture d ecran 2018-12-26 a 09 48 06](https://user-images.githubusercontent.com/623763/50439832-cfecf980-08f3-11e9-90bb-dae92c7f85e7.png)

**Comment by PabloKowalczyk** [comment]
We have answer, thank @kinoute.

**Comment by PabloKowalczyk** [comment]
@pridit @kinoute colud you test that version https://github.com/PabloKowalczyk/crunz/tree/fix-missing-tasks-source? Remember about adding `-vvv` switch, should tell you more and could be useful in further debug/fixing.

**Comment by kinoute** [comment]
Is it helpful?

![capture d ecran 2018-12-27 a 11 53 16](https://user-images.githubusercontent.com/623763/50477770-12830480-09ce-11e9-8b1b-a8e37c655037.png)

**Comment by PabloKowalczyk** [comment]
It is :) Forgot about `tasks` in legacy paths.

**Comment by PabloKowalczyk** [comment]
@kinoute could test again after update, please?

**Comment by kinoute** [comment]
No change :(

![capture d ecran 2018-12-29 a 13 58 31](https://user-images.githubusercontent.com/623763/50538597-fcae4480-0b71-11e9-9f11-a88463ae171e.png)

**Comment by PabloKowalczyk** [comment]
Interesting, where are your tasks? In `/Volumes/Storage/kinoute/Sites/crunz-error/tasks`?

**Comment by kinoute** [comment]
There are in `app/tasks`.

My project's tree is like this:

![capture d ecran 2018-12-29 a 14 06 09](https://user-images.githubusercontent.com/623763/50538669-071d0e00-0b73-11e9-8bef-51c3fd936ddb.png)

crunz.yml

![capture d ecran 2018-12-29 a 14 06 57](https://user-images.githubusercontent.com/623763/50538673-0e441c00-0b73-11e9-920b-8253e67e5f3b.png)

**Comment by PabloKowalczyk** [comment]
So, i must also fallback `config.yml` resolution, thanks @kinoute!

**Comment by PabloKowalczyk** [comment]
Another update pushed, could you test it @kinoute? Maybe @pridit  also have some time to test patch mentioned earlier?
In my tests it fallbacks as expected, if you confirm it also works for you it will be first step to stable `v1.11`.

**Comment by kinoute** [comment]
![capture d ecran 2019-01-06 a 14 37 52](https://user-images.githubusercontent.com/623763/50736681-b2a40f00-11c0-11e9-923c-a01bcdfc24ea.png)

**Comment by PabloKowalczyk** [comment]
Already pushed fixes, could retest please?

**Comment by kinoute** [comment]
No more "no task found" error, but the task fails:

![capture d ecran 2019-01-08 a 21 59 24](https://user-images.githubusercontent.com/623763/50858495-d5851d80-1390-11e9-9958-3e41fda90a11.png)

![capture d ecran 2019-01-08 a 22 00 01](https://user-images.githubusercontent.com/623763/50858507-dcac2b80-1390-11e9-93da-3bd630c4aefb.png)

**Comment by PabloKowalczyk** [comment]
Did that task work in `v1.10`?

**Comment by kinoute** [comment]
the task is only `echo 'test';` 😃

**Comment by PabloKowalczyk** [comment]
Your screenshot tells `cd app/game/test; php test.php` :)

**Comment by kinoute** [comment]
It's because I use the `in()` functionality. Here is the task:

````<?php

require_once dirname(__DIR__) . '/../vendor/autoload.php';

use Crunz\Schedule;

$schedule = new Schedule();

$schedule->run('php test.php')
         ->everyMinute()
         ->in('app/game/test')
         ->description('Test');

return $schedule;

**Comment by PabloKowalczyk** [comment]
Done some testing with similar task:
```
<?php

use Crunz\Schedule;

$scheduler = new Schedule();

$scheduler
    ->run('pwd')
    ->in('vendor/symfony')
    ->description('PWD')
    ->everyMinute();

return $scheduler;
```
`v1.7.3`, `v1.10.1` and `v1.11-rc.1` completed with `success`, but only because `pwd` is global binary. `cd`ing always was `cwd`-relative and this behavior not changed in `v1.11`. Could you test some older versions to make sure `v1.11` not break something?

**Comment by kinoute** [comment]
Same problem with v1.10.1. Do you want me to upload my test repo?

**Comment by PabloKowalczyk** [comment]
No, it is not necessary. Bug seems solved to me, WDYT?

**Comment by kinoute** [comment]
Really? I have still the problem here, even with your last commits on` fix-missing-tasks-source`.

Running vendor/bin/crunz schedule:run -vvv within the repo : works
Simulating the cron job call with /path/to/project/vendor/bin/crunz... makes the task fail.

https://gist.github.com/kinoute/2f1eec3c4908f17b63001aa29bc38795

**Comment by PabloKowalczyk** [comment]
>Running vendor/bin/crunz schedule:run -vvv within the repo : works
Simulating the cron job call with /path/to/project/vendor/bin/crunz... makes the task fail.

Isn't it the same behavior as on `v1.10.1`, `v1.7.3` or any other version? I have tested it and it fails on any earlier version, not only newest `v1.11`. The `->in()` call was always related to `cwd`. If you think that `->in()` should work it is other case which we can discuss in other issue, but it is not caused by changes in `v1.11`.

**Comment by PabloKowalczyk** [comment]
What version you use in production? Maybe I'm wrong and `->in()` is broken only in `v1.11` or maybe commands you use in production are globally and do not need changing `cwd`, so even if `cd` fail they can end successfully? Could you paste one of these commands?

**Comment by kinoute** [comment]
I posted my test repo here: https://github.com/kinoute/crunz-error (used locally on PHP 7.3 by the way, using your latest fixes on your personal GitHub repo). That's the one who works from within the repo, but the task fails when getting called by the cronjob call /path/to/project etc.

To sum up my use of crunz, I've been using the same kind of tasks (all of them using `in()`) since you fixed the CPU usage in [v1.5.1](https://github.com/lavary/crunz/releases/tag/v1.5.1). I started to have description/error missing in email errors afterwards (#161) in v1.10.1 and you fixed the bug not into the same version, but in the next one, v1.11.0-beta.2.

Therefore, I updated my version in production, but then I got the problem described here, "No task found". I immediately downgraded at that time because I couldn't reproduce the bug locally. Since this issue has been discussed here, I updated again to v1.11.0-rc.1 in production. I still have the same issue about tasks not being found, but I use a workaround to avoid this error (I basically changed the cronjob call to "`cd /path/to/project && vendor...`").

I also have a problem where some scripts generate an empty error every time from crunz as discussed in the other issue, but it seems related to a PHP 7.3.0 bug and I'm waiting 7.3.1 or 7.3.2 to see how it goes. For the moment, I disabled `email_errors`.
